File: serviceAccount/src/main/debug/java/com/effective/android/service/account/TestActivity.java
package com.effective.android.service.account;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.effective.android.base.rxjava.Rx2Schedulers;
import com.effective.android.base.toast.ToastUtils;
import com.effective.android.base.util.GsonUtils;
import com.plugin.component.ComponentManager;
import com.plugin.component.SdkManager;

import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;

public class TestActivity extends AppCompatActivity implements AccountChangeListener {

    private UserInfo userInfo;
    private Disposable disposable;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.account_activity_main);
        init();
    }

    private void init() {
        ComponentManager.init(getApplication());
        AccountSdk accountSdk = SdkManager.getSdk(AccountSdk.class);
        userInfo = accountSdk.getAccount();
        checkoutStatus(userInfo != null, userInfo);
        findViewById(R.id.login).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                accountSdk.login(TestActivity.this);
            }
        });
        findViewById(R.id.logout).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                disposable = accountSdk.logout()
                        .compose(Rx2Schedulers.flowableIoToMain())
                        .subscribe(aBoolean -> ToastUtils.show(TestActivity.this, aBoolean ? "已退出登陆了！" : "退出登陆失败！"),
                                throwable -> ToastUtils.show(TestActivity.this, "退出登陆失败！"));
            }
        });
    }

    @Override
    public void onAccountChange(@org.jetbrains.annotations.Nullable UserInfo userInfo, boolean login, boolean success, @org.jetbrains.annotations.Nullable String message) {
        if (success) {
            checkoutStatus(login, userInfo);
        }
    }

    private void checkoutStatus(boolean login, UserInfo userInfo) {
        findViewById(R.id.login).setEnabled(!login);
        findViewById(R.id.logout).setEnabled(login);
        ((TextView) findViewById(R.id.info)).setText(login ? GsonUtils.getJsonString(userInfo) : "");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (disposable != null && !disposable.isDisposed()) {
            disposable.dispose();
        }
    }
}

File: debugModule/src/main/sAccount/java/com/debug/service/account/TestActivity.kt
package com.debug.service.account

import android.os.Bundle
import android.view.View
import android.widget.TextView
import com.debug.R

import com.effective.android.base.activity.BaseActivity
import com.effective.android.base.systemui.StatusbarHelper
import com.effective.android.base.toast.ToastUtils
import com.effective.android.base.util.GsonUtils
import com.effective.android.service.account.AccountChangeListener
import com.effective.android.service.account.AccountSdk
import com.effective.android.service.account.UserInfo
import com.plugin.component.ComponentManager
import com.plugin.component.SdkManager


class TestActivity : BaseActivity(), AccountChangeListener {

    private var userInfo: UserInfo? = null
    private var accountSdk: AccountSdk? = null

    override fun getLayoutRes(): Int {
        return R.layout.account_activity_main_layout
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.account_activity_main_layout)
        StatusbarHelper.translucentStatusBar(this)
        init()
    }

    private fun init() {
        ComponentManager.init(application)
        accountSdk = SdkManager.getSdk(AccountSdk::class.java)
        userInfo = accountSdk!!.getAccount()
        accountSdk!!.addAccountChangeListener(this)
        checkoutStatus(userInfo != null, userInfo)
        findViewById<View>(R.id.login).setOnClickListener { accountSdk!!.login(this@TestActivity) }
        findViewById<View>(R.id.logout).setOnClickListener { accountSdk!!.logout() }
    }

    override fun onAccountChange(userInfo: UserInfo?, login: Boolean, success: Boolean, message: String?) {
        runOnUiThread {
            if (success) {
                checkoutStatus(login, userInfo)
            }
            ToastUtils.show(this@TestActivity, message!!)
        }
    }

    private fun checkoutStatus(login: Boolean, userInfo: UserInfo?) {
        findViewById<View>(R.id.login).isEnabled = !login
        findViewById<View>(R.id.logout).isEnabled = login
        (findViewById<View>(R.id.info) as TextView).text = if (login) GsonUtils.getJsonString(userInfo!!) else ""
    }

    override fun onDestroy() {
        super.onDestroy()
        if (accountSdk != null) {
            accountSdk!!.removeAccountChangeListener(this)
        }
    }
}
